logger.js

```diff
@@ -50,31 +52,52 @@ function Logger(opts) {
 
     this.statsd = opts.statsd;
 
-    var levels = extend(defaultLevels, opts.levels || {});
-    var path = opts.path || "";
+    var levels = this.levels = extend(defaultLevels, opts.levels || {});
+    this.path = opts.path = "";
 
     Object.keys(levels)
-        .forEach(function (levelName) {
+        .forEach(function makeMethodForLevel(levelName) {
             if (!levels[levelName]) {
                 return;
             }
 
             var level = levels[levelName];
             level = extend({transforms: []}, level);
+            levels[levelName] = level;
+
             level.transforms = level.transforms.concat(transforms);
 
-            this[levelName] = logMethod(this, levelName, level, path);
-        }, this);
+            self[levelName] = makeLogMethod(levelName);
+        }, {});
+
+    this._streamsByLevel = Object.keys(levels)
+        .reduce(function accumulateStreamsByLevel(streamsByLevel, levelName) {
+            if (!levels[levelName]) {
```

Comment:
nitpick. I don't know why you don't yank the level declaration and just use that.